### PR TITLE
Language screen

### PR DESCRIPTION
**Pull request by pisarevkir**

### CODE REVIEW

File: src/navigation/components/Header/components/LangButton.tsx [hunk 1]
```diff
@@ -29,13 +30,15 @@ export const LangButton = ({
...
+    useNavigation<NativeStackNavigationProp<ProfileTabStackParamList>>();
   return (
     <TouchableOpacity
-      onPress={() => navigation.navigate('HomeTab')}
+      onPress={() => navigation.navigate('LanguageSettings')}
       hitSlop={SMALL_BUTTON_HIT_SLOP}>
       <View style={[styles.container, containerStyle]}>
-        <Text style={[styles.langText, {color}]}>EN</Text>
+        <Text style={[styles.langText, {color}]}>
+          {t('global.language_code').toUpperCase()}
```
Comment (victor-oancea-leftclick-io) [inline src/navigation/components/Header/components/LangButton.tsx:40]: the code is not translateable

File: src/screens/SettingsFlow/LanguageSettings/components/LanguageListItem.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  ({language, selected, loading, onSelect}: Props) => {
+    return (
+      <View style={styles.container}>
+        <Text
+          style={[
+            styles.languageText,
+            selected && styles.languageText_selected,
+          ]}
+          numberOfLines={1}>
+          {t('global.language', {locale: language})}
```
Comment (victor-oancea-leftclick-io) [inline src/screens/SettingsFlow/LanguageSettings/components/LanguageListItem.tsx:28]: you need to include the language in the key: global.language.$language

File: src/screens/SettingsFlow/LanguageSettings/index.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        contentContainerStyle={[commonStyles.shadow, bottomOffset.current]}
+        onScroll={scrollHandler}
+        scrollEventThrottle={16}
+        showsVerticalScrollIndicator={false}>
+        <View style={styles.listContent}>
+          {availableLocales.map((language, index) => (
+            <React.Fragment key={language}>
+              {index !== 0 && <LanguageListItemSeparator />}
+              <LanguageListItem
+                selected={deviceSettings?.language === language}
```
Comment (victor-oancea-leftclick-io) [inline src/screens/SettingsFlow/LanguageSettings/index.tsx:54]: case insensitive equals pls

File: src/store/modules/Devices/sagas/updateDeviceSettings.ts [hunk 4]
```diff
@@ -54,6 +57,10 @@ export function* updateDeviceSettingsSaga(
       settings,
     });
     yield put(DeviceActions.UPDATE_SETTINGS.SUCCESS.create(updatedSetings));
+    if (settings.language) {
+      I18nManager.forceRTL(t('isRTL', {locale: settings.language}) === 'true');
```
Comment (victor-oancea-leftclick-io) [inline src/store/modules/Devices/sagas/updateDeviceSettings.ts:61]: whats rtl?

File: src/screens/SettingsFlow/LanguageSettings/components/LanguageListItem.tsx (hunk 2 above)
Comment (pisarevkir) [reply src/screens/SettingsFlow/LanguageSettings/components/LanguageListItem.tsx:28]: Didn't get your idea. Now every locale contains it's own language name translated on it's own language and this is exactly what we need for the list

File: src/store/modules/Devices/sagas/updateDeviceSettings.ts (hunk 4 above)
Comment (pisarevkir) [reply src/store/modules/Devices/sagas/updateDeviceSettings.ts:61]: Right-to-left interface, some locales use it -> try to change the lang to RO in the app, i set RTL for Romanian lang for the testing purposes

File: src/screens/SettingsFlow/LanguageSettings/components/LanguageListItem.tsx (hunk 2 above)
Comment (victor-oancea-leftclick-io) [reply src/screens/SettingsFlow/LanguageSettings/components/LanguageListItem.tsx:28]: you have N languages, N translations, yet there is only 1 key `global.language`

File: src/navigation/components/Header/components/LangButton.tsx (hunk 1 above)
Comment (pisarevkir) [reply src/navigation/components/Header/components/LangButton.tsx]: done

**Review by OlehPanforov — APPROVED** [review]

**Review by subbotnik16 — APPROVED** [review]

**Pull request merged**